app/models/similarity_metric.rb

```diff
@@ -132,8 +132,11 @@ def cull_by_zero_crossing( score_map, just_score=true )
   # take a merged set of score maps and calculate a new score using the best weight for each metric
   def rescore_amalgam(score_map)
     score_map = score_map[:metrics_with_scores]
-    return Float::MAX if score_map.empty?
-    score_map.map{ |k,v| k.weight * v }.sum / score_map.size
+    # if we have no data, return a score representing maximal dissimilarity
+    return Float::INFINITY if score_map.empty?
+
+    # return a weighted average
+    score_map.sum { |k,v| k.weight * v } / score_map.keys.sum(&:weight)
```

Comment:
I figured to prevent confusion we should use the same "number" to indicate uselessness everywhere. We were using `Float::MAX` in one place and `Float::INFINITY` in other. I thought the latter was clearer, so I unified on that.

Also, I'm pretty sure this is supposed to be a weighted mean, but we were ignoring the weights in the denominator. I fixed that.